ArgoProject/src/parameters/CameraParameters.h

```diff
@@ -0,0 +1,37 @@
+﻿#ifndef _PARAMETERS_CAMERA_H
+#define _PARAMETERS_CAMERA_H
+
+namespace app::par
+{
+	struct CameraParameters
+	{
+	public: // Constructors/Destructor/Assignments
+		CameraParameters() = default;
+		~CameraParameters() = default;
+
+		CameraParameters(CameraParameters const &) = default;
+		CameraParameters & operator=(CameraParameters const &) = default;
+
+		CameraParameters(CameraParameters &&) = default;
+		CameraParameters & operator=(CameraParameters &&) = default;
+
+	public: // Public Static Functions
+	public: // Public Member Functions
+	public: // Public Static Variables
+	public: // Public Member Variables
+		std::optional<app::Entity> targetEntity = std::nullopt;
+		std::optional<math::Rectf> clampRect;
+		std::optional<bool> clamp = false;
+		std::optional<float> internalClampSize;
```

Comment:
```suggestion
		std::optional<math::Vector2f> internalClamp;
```
internal clamp needs varying x-axis and y-axis values.